Libs/DICOM/Widgets/ctkDICOMBrowser.cpp

```diff
@@ -1749,3 +1749,52 @@ void ctkDICOMBrowser::onIndexingComplete(int patientsAdded, int studiesAdded, in
   // allow users of this widget to know that the process has finished
   emit directoryImported();
 }
+
+//----------------------------------------------------------------------------
+void ctkDICOMBrowser::setSelectedItems(ctkDICOMModel::IndexType level, QStringList uids)
+{
+  Q_D(ctkDICOMBrowser);
+  if (level == ctkDICOMModel::PatientType)
+  {
+    d->dicomTableManager->setCurrentPatientsSelection(uids);
+  }
+  else if (level == ctkDICOMModel::StudyType)
+  {
+    // Select parent patient to make sure the requested studies
+    // are listed in the study table
+    QStringList patientUids;
+    for (QString uid : uids)
```

Comment:
For correctness:

```suggestion
    for (const QString& uid : uids)
```

https://stackoverflow.com/questions/15176104/range-based-loop-get-item-by-value-or-reference-to-const/15176127#15176127